packages/flutter/lib/src/widgets/text_editing_action.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'actions.dart';
+import 'editable_text.dart';
+import 'focus_manager.dart';
+import 'framework.dart';
+
+// Similar to CallbackAction's OnInvokeCallback, but includes EditableTextState.
+typedef _OnInvokeTextEditingCallback<T extends Intent> = Object? Function(T intent, EditableTextState editableTextState);
+
+/// An [Action] related to editing text.
+///
+/// If an [EditableText] is currently focused, then _editableTextState will be
+/// set and the given [onInvoke] callback will be called with it. If not, then
+/// [isEnabled] will be false and [onInvoke] will not be called.
+///
+/// The focused [EditableText] must have a [Key]. This is handled automatically
+/// by built-in text editing widgets like [TextField], [CupertinoTextField],
+/// and [SelectableText], but for custom usage of [EditableText], it is
+/// necessary to explicitly pass in a key.
+///
+/// See also:
+///
+///  * [CallbackAction], which is a similar Action type but unrelated to text
+///    editing.
+class TextEditingAction<T extends Intent> extends Action<T> {
+  /// A constructor for a [TextEditingAction].
+  ///
+  /// The [onInvoke] parameter must not be null.
+  TextEditingAction({required this.onInvoke}) : assert(onInvoke != null);
+
+  EditableTextState? get _editableTextState {
+    // If an EditableText is not focused, then ignore this action.
+    if (primaryFocus?.context?.widget is! EditableText) {
+      return null;
+    }
+    final EditableText editableText = primaryFocus!.context!.widget as EditableText;
```

Comment:
if `context.widget is EditableText` then `(context as StatefulElement).state` must be the EditableTextState. So technically, you wouldn't need a key, I think.

The `is type` checks do seem a little iffy here, though... Do we already have an established pattern somewhere for only doing an action if focus is on a particular widget?

Reply:
The DirectionalFocusIntent seems to do it the same way.  CC @gspencergoog 

https://github.com/flutter/flutter/blob/f49956598b2a44cedcb70bd8c4b6b7ef5dc54d43/packages/flutter/lib/src/widgets/focus_traversal.dart#L1946

I'll take your suggestion of not using the key for now.

Reply:
That usage in the `DirectionFocusIntent` was a workaround for before we had the ability to override key bindings, and the ignoreTextFields field will probably be deprecated now that we can. The `is` check was iffy at the time, but was the only solution we could think of.